devicetypes/smartthings/fibaro-heat-controller.src/fibaro-heat-controller.groovy

```diff
@@ -0,0 +1,334 @@
+		capability "Thermostat Mode"
+		capability "Refresh"
+		capability "Battery"
+		capability "Thermostat Heating Setpoint"
+		capability "Health Check"
+		capability "Thermostat"
+		capability "Temperature Measurement"
+
+		command "setThermostatSetpointUp"
+		command "setThermostatSetpointDown"
+		command "switchMode"
+
+		fingerprint mfr: "010F", prod: "1301", model: "1000", deviceJoinName: "Fibaro Heat Controller"
+	}
+
+	tiles(scale: 2) {
+		multiAttributeTile(name:"thermostat", type:"general", width:6, height:4, canChangeIcon: false)  {
+			tileAttribute("device.heatingSetpoint", key: "VALUE_CONTROL") {
+				attributeState("VALUE_UP", action: "setThermostatSetpointUp")
+				attributeState("VALUE_DOWN", action: "setThermostatSetpointDown")
+			}
+			tileAttribute("device.thermostatMode", key: "PRIMARY_CONTROL") {
+				attributeState("off", action:"switchMode", nextState:"...", icon: "st.thermostat.heating-cooling-off")
+				attributeState("heat", action:"switchMode", nextState:"...", icon: "st.thermostat.heat")
+				attributeState("emergency heat", action:"switchMode", nextState:"...", icon: "st.thermostat.emergency-heat")
+			}
+			tileAttribute("device.temperature", key: "SECONDARY_CONTROL") {
+				attributeState("temperature", label:'${currentValue}°', icon: "st.alarm.temperature.normal",
+						backgroundColors:[
+								// Celsius
+								[value: 0, color: "#153591"],
+								[value: 7, color: "#1e9cbb"],
+								[value: 15, color: "#90d2a7"],
+								[value: 23, color: "#44b621"],
+								[value: 28, color: "#f1d801"],
+								[value: 35, color: "#d04e00"],
+								[value: 37, color: "#bc2323"],
+								// Fahrenheit
+								[value: 40, color: "#153591"],
+								[value: 44, color: "#1e9cbb"],
+								[value: 59, color: "#90d2a7"],
+								[value: 74, color: "#44b621"],
+								[value: 84, color: "#f1d801"],
+								[value: 95, color: "#d04e00"],
+								[value: 96, color: "#bc2323"]
+						]
+				)
+			}
+		}
+
+		valueTile("battery", "device.battery", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "battery", label: 'Battery:\n${currentValue}%', unit: "%"
+		}
+
+		standardTile("refresh", "command.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "refresh", label: 'refresh', action: "refresh.refresh", icon: "st.secondary.refresh-icon"
+		}
+		main "thermostat"
+		details(["thermostat", "battery", "refresh"])
+	}
+}
+
+def installed() {
+	runIn(2, "initialize")
+}
+
+def updated() {
+	initialize()
+}
+
+def initialize() {
+	def supportedModes = ["off", "emergency heat", "heat"]
+	state.supportedModes = supportedModes
+	sendEvent(name: "temperature", value: 0)
+	sendEvent(name: "supportedThermostatModes", value: supportedModes, displayed: false)
+	sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+	response(refresh())
```

Comment:
For functions that are called by runIn, you'll need to use sendHubCommand. Also, the temperature will get reset to 0 every time update() is called. I would use these functions:

```groovy
def installed() {
	log.debug "installed()"
	state.supportedModes = ["off", "emergency heat", "heat"]

	sendEvent(name: "temperature", value: 0, unit: "C", displayed: false)
	sendEvent(name: "supportedThermostatModes", value: state.supportedModes, displayed: false)

	runIn(2, "initialize", [overwrite: true])
}

def updated() {
	log.debug "updated()"
	runIn(2, "initialize", [overwrite: true])
}

def initialize() {
	log.debug "initialize()"

	sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])

	runIn(5, "forcedRefresh", [overwrite: true])
}
```

Reply:
Actually.... This would be better:

```groovy
def installed() {
	log.debug "installed()"
	state.supportedModes = ["off", "emergency heat", "heat"]

	sendEvent(name: "temperature", value: 0, unit: "C", displayed: false)
	sendEvent(name: "supportedThermostatModes", value: state.supportedModes, displayed: false)

	runIn(2, "updated", [overwrite: true])
}

def updated() {
	log.debug "updated()"

	sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])

	runIn(5, "forcedRefresh", [overwrite: true])
}
```